### PR TITLE
chore: Improve charm config_changed event handler method

### PR DESCRIPTION
**Pull request by gatici**
# Description

- Improves ConfigChanged event handling process by calling main callback (self._configure_webui)
- Removes unnecessary doc strings from some private methods

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

**Review by Gmerold — CHANGES_REQUESTED** [review]

File: src/charm.py [hunk 1]
```diff
@@ -112,41 +112,39 @@ def __init__(self, *args):
         self.framework.observe(self._common_database.on.endpoints_changed, self._configure_webui)
         self.framework.observe(self._auth_database.on.endpoints_changed, self._configure_webui)
         self.framework.observe(
-            self.on.sdcore_management_relation_joined, self._publish_sdcore_management_url
+            self.on.sdcore_management_relation_joined, self._on_sdcore_management_relation_joined
```
[the remark this answers is not in the document]
Comment (gatici) [reply src/charm.py]: Used only configure_webui method for all Juju events.

File: src/charm.py [hunk 2]
```diff
@@ -112,41 +112,39 @@ def __init__(self, *args):
...
         # Handling config changed event to publish the new url if the unit reboots and gets new IP
-        self.framework.observe(self.on.config_changed, self._publish_sdcore_management_url)
+        self.framework.observe(self.on.config_changed, self._configure_webui)
 
     def _configure_webui(self, event: EventBase) -> None:
-        """Handle config changes.
+        """Handle the config changes.
 
-        Manage pebble layer and Juju unit status.
+        Manage pebble layer and Juju unit status because this is the main callback method.
```
[the remark this answers is not in the document]
Comment (gatici) [reply src/charm.py]: The method docstring is updated.

File: src/charm.py [hunk 3]
```diff
@@ -112,41 +112,40 @@ def __init__(self, *args):
...
+            self.on.sdcore_management_relation_joined, self._configure_webui
         )
         # Handling config changed event to publish the new url if the unit reboots and gets new IP
-        self.framework.observe(self.on.config_changed, self._publish_sdcore_management_url)
+        self.framework.observe(self.on.config_changed, self._configure_webui)
 
-    def _configure_webui(self, event: EventBase) -> None:
-        """Handle config changes.
+    def _configure_webui(self, _: EventBase) -> None:
+        """Configure Webui configuration file and publishes the Webui management URL.
```
Comment (Gmerold) [inline src/charm.py:121]:
```suggestion
        """The main callback method for Juju events.
```

File: src/charm.py [hunk 4]
```diff
@@ -195,8 +194,28 @@ def _on_collect_unit_status(self, event: CollectStatusEvent):
...
 
+    def _generate_webui_config_file(self) -> str:
+        return render_config_file(
+            common_database_name=COMMON_DATABASE_NAME,
+            common_database_url=self._get_common_database_url(),
+            auth_database_name=AUTH_DATABASE_NAME,
+            auth_database_url=self._get_auth_database_url(),
+        )
+
+    def _configure_workload(self, restart: bool = False) -> None:
```
Comment (Gmerold) [inline src/charm.py:205]: Personally I'd write some docstring here explaining the conditions.

File: src/charm.py [hunk 5]
```diff
@@ -112,41 +112,40 @@ def __init__(self, *args):
...
+        self.framework.observe(self.on.config_changed, self._configure_webui)
 
-    def _configure_webui(self, event: EventBase) -> None:
-        """Handle config changes.
+    def _configure_webui(self, _: EventBase) -> None:
+        """Configure Webui configuration file and publishes the Webui management URL.
 
-        Manage pebble layer and Juju unit status.
+        The main callback method for `config changed`, `pebble ready`, `relation changed` e.g.
+         and custom events. Manages the Pebble services.
```
Comment (Gmerold) [inline src/charm.py:124]:
```suggestion
        Whenever a Juju event is emitted, this method is called to configure the Webui workload, run it and expose the service information through charm's interface. Prior to running any of the above, the method performs a couple of checks to make sure that the workload is ready to be started. 
```
How 'bout something along these lines?

File: src/charm.py (hunk 3 above)
Comment (gatici) [reply src/charm.py:121]: > 

The first line of the docstring should be imperative. "The main callback method for Juju events." is not acceptable in format checking.

Comment (gatici) [reply src/charm.py]: The docstring is updated.

File: src/charm.py (hunk 4 above)
Comment (gatici) [reply src/charm.py:206]: A docstring is added to explain the conditions.

File: src/charm.py (hunk 5 above)
Comment (gatici) [reply src/charm.py]: Done

**Review by Gmerold — APPROVED** [review]

**Pull request merged**